# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/Selection.h
# Commit: 71670cedaf26
# Full Hash: 71670cedaf26214f3bcb8562a445cf1ef24f4db6
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-06-30 09:36:48
# Regressor Bug: 1714914
# File Overlap Count: 1
# Description:
#   Bug 1714914 - Make editor classes treat offset in parent node as uint32_t r=m_kato
#   
#   It should be treated as `uint32_t` since DOM API does so.  However, there are
#   some exceptions:
#   
# ==============================================================================

diff -r f5dcc2d793d6 -r 71670cedaf26 dom/base/Selection.h
--- a/dom/base/Selection.h	Tue Jun 29 22:00:09 2021 +0000
+++ b/dom/base/Selection.h	Wed Jun 30 07:07:28 2021 +0000
@@ -193,7 +193,7 @@
   nsresult RemoveCollapsedRanges();
   void Clear(nsPresContext* aPresContext);
   MOZ_CAN_RUN_SCRIPT nsresult CollapseInLimiter(nsINode* aContainer,
-                                                int32_t aOffset) {
+                                                uint32_t aOffset) {
     if (!aContainer) {
       return NS_ERROR_INVALID_ARG;
     }
@@ -460,7 +460,7 @@
    * in the given node. When the selection is collapsed, and the content
    * is focused and editable, the caret will blink there.
    * @param aContainer The given node where the selection will be set
-   * @param offset      Where in given dom node to place the selection (the
+   * @param aOffset     Where in given dom node to place the selection (the
    *                    offset into the given node)
    */
   MOZ_CAN_RUN_SCRIPT void CollapseInLimiter(nsINode& aContainer,